system_baseline/models.py

```diff
@@ -176,3 +176,21 @@ def get_mapped_system_count(cls, account_number, org_id, rbac_group_filters=None
             results = query.filter(cls.account == account_number).all()
 
         return results
+
+    @classmethod
+    def update_systems(cls, system_id, groups=None):
+        query = SystemBaselineMappedSystem.query.filter(
+            SystemBaselineMappedSystem.system_id == system_id
+        )
+        systems = query.all()
+        for system in systems:
+            if groups is not None:
```

Comment:
In case of `groups` of `None` this would be a noop, hence no loop, nor query, nor commit is needed. Unless I've missed something.